shell/platform/windows/flutter_windows_engine_unittests.cc

```diff
@@ -95,6 +96,63 @@ TEST(FlutterWindowsEngine, RunDoesExpectedInitialization) {
   modifier.embedder_api().Shutdown = [](auto engine) { return kSuccess; };
 }
 
+TEST(FlutterWindowsEngine, RunWithoutANGLEUsesSoftware) {
+  std::unique_ptr<FlutterWindowsEngine> engine = GetTestEngine();
+  EngineEmbedderApiModifier modifier(engine.get());
+
+  // The engine should be run with expected configuration values.
+  bool run_called = false;
+  modifier.embedder_api().Run = MOCK_ENGINE_PROC(
+      Run, ([&run_called, engine_instance = engine.get()](
+                size_t version, const FlutterRendererConfig* config,
+                const FlutterProjectArgs* args, void* user_data,
+                FLUTTER_API_SYMBOL(FlutterEngine) * engine_out) {
+        run_called = true;
+        *engine_out = reinterpret_cast<FLUTTER_API_SYMBOL(FlutterEngine)>(1);
+        // We don't have an AngleSurfaceManager, so we should be using software.
+        EXPECT_EQ(config->type, kSoftware);
+        return kSuccess;
+      }));
+
+  // It should send locale info.
+  bool update_locales_called = false;
+  modifier.embedder_api().UpdateLocales = MOCK_ENGINE_PROC(
+      UpdateLocales,
+      ([&update_locales_called](auto engine, const FlutterLocale** locales,
+                                size_t locales_count) {
+        update_locales_called = true;
+
+        EXPECT_GT(locales_count, 0);
+        EXPECT_NE(locales, nullptr);
+
+        return kSuccess;
+      }));
```

Comment:
I don't see any problem testing this, but is it likely this doesn't get called?

Reply:
I don't think so. I've removed that test as it's already tested in the prior unit test, and left the stubs for those Embedder API methods. They need to be kept as stubs as we don't do enough actual initialisation to support calling those APIs properly, and they're not relevant to this unit test anyway.